xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-test/xwiki-platform-administration-test-docker/src/test/it/org/xwiki/administration/test/ui/RegisterIT.java

```diff
@@ -0,0 +1,301 @@
+        assertFalse(validateAndRegister(testUtils, useLiveValidation, isModal, registrationPage));
+        assertTrue(registrationPage.validationFailureMessagesInclude("The two passwords do not match."));
+    }
+
+    @ParameterizedTest
+    @MethodSource("testsParameters")
+    @Order(5)
+    void registerEmptyPassword(boolean useLiveValidation, boolean isModal, TestUtils testUtils) throws Exception
+    {
+        AbstractRegistrationPage registrationPage = setUp(testUtils, useLiveValidation, isModal);
+        registrationPage.fillRegisterForm(null, null, null, "", "", null);
+        assertFalse(validateAndRegister(testUtils, useLiveValidation, isModal, registrationPage));
+        assertTrue(registrationPage.validationFailureMessagesInclude("This field is required."));
+    }
+
+    @ParameterizedTest
+    @MethodSource("testsParameters")
+    @Order(6)
+    void registerEmptyUserName(boolean useLiveValidation, boolean isModal, TestUtils testUtils) throws Exception
+    {
+        AbstractRegistrationPage registrationPage = setUp(testUtils, useLiveValidation, isModal);
+        // A piece of javascript fills in the username with the first and last names so we will empty them.
+        registrationPage.fillRegisterForm("", "", "", null, null, null);
+        assertFalse(validateAndRegister(testUtils, useLiveValidation, isModal, registrationPage));
+        assertTrue(registrationPage.validationFailureMessagesInclude("This field is required."));
+    }
+
+    @ParameterizedTest
+    @MethodSource("testsParameters")
+    @Order(7)
+    void registerInvalidEmail(boolean useLiveValidation, boolean isModal, TestUtils testUtils) throws Exception
+    {
+        AbstractRegistrationPage registrationPage = setUp(testUtils, useLiveValidation, isModal);
+        registrationPage.fillRegisterForm(null, null, null, null, null, "not an email address");
+        assertFalse(validateAndRegister(testUtils, useLiveValidation, isModal, registrationPage));
+        assertTrue(registrationPage.validationFailureMessagesInclude("Please enter a valid email address."));
+    }
+
+    private AbstractRegistrationPage setUp(TestUtils testUtils, boolean useLiveValidation, boolean isModal)
+        throws Exception
+    {
+        // We create the admin user because it is expected to exist when testing the registration of an existing user.
+        testUtils.loginAsSuperAdmin();
+        testUtils.createAdminUser();
+        deleteJohnSmith(testUtils);
+        testUtils.updateObject("XWiki", "RegistrationConfig", "XWiki.Registration", 0, "liveValidation_enabled",
+            useLiveValidation);
+        switchUser(testUtils, isModal);
+        testUtils.recacheSecretToken();
+        AbstractRegistrationPage registrationPage = this.getRegistrationPage(isModal);
+        // The prepareName javascript function is the cause of endless flickering since it tries to suggest a username
```

Comment:
That's an old comment or something you checked yourself? I don't really understand this "tries to suggest a username" that's something still accurate?

Reply:
It's an old comment, not fully sure it is still accurate though.

Reply:
ok might be good to at least add a TODO for checking that.

Reply:
`prepareName` is still existing in `xwiki.js`, probably still good to have a warning not to use it even if the reason was lot in the migration from XE.